llm_compmat/backend.py

```diff
@@ -0,0 +1,46 @@
+import os
+
+try:
+    from langchain_openai import ChatOpenAI
+    openai_backend_available = True
+except ImportError:
+    openai_backend_available = False
+
+try:
+    from langchain_groq import ChatGroq
+    groq_backend_available = True
+except ImportError:
+    groq_backend_available = False
+
+
+backend_lst = [openai_backend_available, groq_backend_available]
+
+
+if not any(backend_lst):
+    raise ImportError(
+        "None of the supported backends is installed. Install one of the following: langchain_openai, langchain_groq"
+    )
+
+
+def get_backend(openapi_token=None, groq_token=None, **kwargs):
```

Comment:
what about `token` and `provider` rather than one variable per provider that we want to add?

Reply:
Might make it a less flexible for the user, but if we support (or plan to) even more backends I would also change to that format